components/RHPContainer.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import {Pressable, View, Dimensions} from 'react-native';
+const isSmallScreenWidth = Dimensions.get('window').width <= 800;
+
+const RHPContainer = ({navigation, children, contentStyle}) => {
+    return (
+        <View style={{flexDirection: 'row', width: '100%', height: '100%'}}>
+            {!isSmallScreenWidth && <Pressable style={{ flex: 1 }} onPress={() => navigation.goBack()}>
```

Comment:
**noting:** If the "small screens" use the native navigator then we don't need this check

Reply:
I am not sure `goBack()` is what we want for this case (but maybe it is). Just making sure that this does a "dismiss" action and will always pop the entire RHP stack and not just a single screen in the stack.

Reply:
Navigation seems to work here just fine with both `up` and `back` action.